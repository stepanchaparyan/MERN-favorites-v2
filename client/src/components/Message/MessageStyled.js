import styled, { keyframes } from 'styled-components';

export const Container = styled.div`
  padding: 6px 0;
  background-color: lightGray;
  color: black;
  width: 100%;
  height: 16px;
  position: absolute;
  top: 55px;
  font-size: 16px;
  text-align: center;
  display: inline-block;
  animation: ${() => fade} 10s linear;
`;

const fade = keyframes`
  0%{
    opacity: 0;
  }
  50%{
    opacity: 1;
  }
  100%{
    opacity: 0;
  }
`;
