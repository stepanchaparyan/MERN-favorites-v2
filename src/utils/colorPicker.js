import theme from '../styles/theme';
import { FAVITEM } from '../constants';

const { red, blue, limeGreen, orange, black } = theme;
const { TYPE } = FAVITEM;

export const colorPicker = category => {
  let color;
  switch (category) {
    case TYPE.FILM:
      color = red;
      break;
    case TYPE.MUSIC:
      color = blue;
      break;
    case TYPE.BOOKS:
      color = limeGreen;
      break;
    case TYPE.OTHER:
      color = orange;
      break;
    default:
      color = black;
  }
  return color;
};
