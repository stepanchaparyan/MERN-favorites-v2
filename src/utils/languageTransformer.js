const dict = {
  // lang - country
  ar: 'sa',
  be: 'bl',
  cs: 'sz',
  da: 'dk',
  de: 'de',
  el: 'gr',
  en: 'us',
  es: 'es',
  fa: 'ir',
  fr: 'fr',
  ja: 'jp',
  hi: 'in',
  hu: 'hu',
  hr: 'hr',
  hy: 'am',
  it: 'it',
  kk: 'kz',
  nl: 'nl',
  no: 'no',
  pl: 'pl',
  pt: 'pt',
  ru: 'ru',
  sk: 'sk',
  sl: 'sl',
  sv: 'se',
  tr: 'tr',
  uk: 'ua'
};

export const languageTransformer = listList => {
  let newList = [];
  listList.map(lang => {
    const item = Object.keys(dict).find(item => item === lang);
    if (item) {
      newList.push(dict[item].toUpperCase());
    }
  });
  return newList;
};

function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export const setLanguage = (code, setLocale) => {
  const countryCode = Object.values(dict).includes(code.toLowerCase());

  if (countryCode) {
    setLocale(getKeyByValue(dict, code.toLowerCase()));
  }
};
