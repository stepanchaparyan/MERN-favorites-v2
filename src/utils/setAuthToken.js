import axios from 'axios';

const Authorization = 'Authorization';

const setAuthToken = token => {
  if (token) {
    axios.defaults.headers.common = { Authorization: `Bearer ${token}` };
  } else {
    delete axios.defaults.headers.common[Authorization];
  }
};
export default setAuthToken;
