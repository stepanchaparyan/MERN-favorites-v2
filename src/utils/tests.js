import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
