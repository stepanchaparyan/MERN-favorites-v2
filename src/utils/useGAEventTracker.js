// eslint-disable-next-line no-unused-vars
import React from 'react';
import ReactGA from 'react-ga';

const useGAEventTracker = (category = 'Category') => {
  return (action = 'action', label = 'label') => {
    ReactGA.event({
      category,
      action,
      label,
    });
  };
};

export default useGAEventTracker;
