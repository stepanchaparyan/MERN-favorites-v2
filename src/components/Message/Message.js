import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './MessageStyled';

const Message = ({ msg }) => {
  return <Container>{msg}</Container>;
};

Message.propTypes = {
  msg: PropTypes.string.isRequired
};

export default Message;
