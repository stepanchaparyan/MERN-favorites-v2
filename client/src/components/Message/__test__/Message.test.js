import { renderSnapshot } from '../../../utils/tests';
import Message from '../Message.js';

const msg = 'testMessage';

describe('Message component test with Enzyme', () => {
  test('renders correct snapshot', () => {
    renderSnapshot(Message, { msg });
  });
});
