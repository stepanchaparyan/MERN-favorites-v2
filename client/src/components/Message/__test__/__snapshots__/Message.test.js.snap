// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component test with Enzyme renders correct snapshot 1`] = `
<styled.div>
  testMessage
</styled.div>
`;
